integration-test-groups/azure/azure-storage-blob/src/test/java/org/apache/camel/quarkus/component/azure/storage/blob/it/AzureStorageBlobTest.java

```diff
@@ -63,47 +87,443 @@ private static BlobContainerClient blobContainer() {
-                .post("/azure-storage-blob/blob/create")
-                .then()
-                .statusCode(201);
-
-        // Read
-        RestAssured.get("/azure-storage-blob/blob/read")
-                .then()
-                .statusCode(200)
-                .body(is(blobContent));
-
-        // Update
-        String updatedContent = blobContent + " updated";
-        RestAssured.given()
-                .contentType(ContentType.TEXT)
-                .body(updatedContent)
-                .patch("/azure-storage-blob/blob/update")
-                .then()
-                .statusCode(200);
-
-        RestAssured.get("/azure-storage-blob/blob/read")
-                .then()
-                .statusCode(200)
-                .body(is(updatedContent));
-
-        // Delete
-        RestAssured.delete("/azure-storage-blob/blob/delete")
-                .then()
-                .statusCode(204);
+        try {
+            // Create
+            RestAssured.given()
+                    .contentType(ContentType.TEXT)
+                    .body(BLOB_CONTENT)
+                    .post("/azure-storage-blob/blob/create")
+                    .then()
+                    .statusCode(201);
+
+            // Read
+            RestAssured.get("/azure-storage-blob/blob/read")
+                    .then()
+                    .statusCode(200)
+                    .body(is(BLOB_CONTENT));
+
+            // List
+            RestAssured.get("/azure-storage-blob/blob/list")
+                    .then()
+                    .statusCode(200)
+                    .body("blobs[0].name", is(AzureStorageBlobRoutes.BLOB_NAME));
+
+            // Update
+            String updatedContent = BLOB_CONTENT + " updated";
+            RestAssured.given()
+                    .contentType(ContentType.TEXT)
+                    .body(updatedContent)
+                    .patch("/azure-storage-blob/blob/update")
+                    .then()
+                    .statusCode(200);
+
+            RestAssured.get("/azure-storage-blob/blob/read")
+                    .then()
+                    .statusCode(200)
+                    .body(is(updatedContent));
+        } finally {
+            // Delete
+            RestAssured.delete("/azure-storage-blob/blob/delete")
+                    .then()
+                    .statusCode(204);
+        }
+    }
+
+    @Test
+    public void download() throws IOException {
+        Path path = null;
+        try {
+            // Create
+            RestAssured.given()
+                    .contentType(ContentType.TEXT)
+                    .body(BLOB_CONTENT)
+                    .post("/azure-storage-blob/blob/create")
+                    .then()
+                    .statusCode(201);
+
+            // Download file
+            String downloadPath = RestAssured.get("/azure-storage-blob/blob/download")
+                    .then()
+                    .statusCode(200)
+                    .body(endsWith("target/test"))
```

Comment:
Yeah, good spot. It would fail because the REST endpoint returns the absolute file path, which on Windows will naturally have different separators. So it'll need 'nixifying' before being returned and the test assertion can remain as-is.